src/libs/IOUUtils.js

```diff
@@ -36,10 +45,15 @@ function calculateAmount(participants, total, isDefaultUser = false) {
  * @param {Object} iouReport
  * @param {String} actorEmail
  * @param {Number} amount
+ * @param {String} currency
  * @param {String} type
  * @returns {Object}
  */
-function updateIOUOwnerAndTotal(iouReport, actorEmail, amount, type = CONST.IOU.REPORT_ACTION_TYPE.CREATE) {
+function updateIOUOwnerAndTotal(iouReport, actorEmail, amount, currency, type = CONST.IOU.REPORT_ACTION_TYPE.CREATE) {
+    if (currency !== iouReport.currency && isNetworkOffline) {
```

Comment:
I don't think this should be only if the user is offline. If I request money in different currency even when being online, I will create optimistic report which will change the total locally to wrong number before this will be fetched from the server to the correct amount.

Reply:
Yeah i think this could end up with the flickering issues.

cc @youssef-lr maybe something we should address in the related issues